compiler/resolution/lifetime.cpp

```diff
@@ -37,12 +37,18 @@
    simple value variable (e.g. an integer) has lifetime of the function block it
    is declared in.
 
-   The reachability of a symbol is simply the lexical block in which
-   it is alive.
+   The checker works with two kinds of lifetimes for each symbol:
+     * intrinsic lifetime, based upon the scope of the variable
+       and when it will be destroyed. For references, the intrinsic lifetime
+       is set to the lifetime of whatever the reference points to
+       during the one moment it is initialized.
+     * inferred lifetime, based on what that symbol is set to
+       (e.g. for a variable of borrow class type, this is the minimum
+        of the lifetimes stored in it)
 
    The checker emits errors for several cases:
 
-     * reachability of a symbol is greater than its lifetime
+     * intrinsic lifetime of a symbol is greater than its inferred lifetime
      * setting a symbol to another value with a shorter lifetime
```

Comment:
The lifetime terminology is still not clicking for me.
I think you change is a step forward, anyway.